src/System.Net.WebSockets.Client/tests/ConnectTest.cs

```diff
@@ -29,7 +28,12 @@ public class ConnectTest : ClientWebSocketTestBase
 
                 Assert.Equal(WebSocketError.Success, ex.WebSocketErrorCode);
                 Assert.Equal(WebSocketState.Closed, cws.State);
-                Assert.Equal(ResourceHelper.GetExceptionMessage("net_webstatus_ConnectFailure"), ex.Message);
+
+                // The .Net Native toolchain optimizes away exception messages.
+                if (!PlatformDetection.IsNetNative)
```

Comment:
Changed to `PlatformDetection` instead of `AssertExtension`, because we also want to validate the `ex.WebSocketErrorCode`.

Reply:
You should still be able to use AssertExtension.Throws<T> and save the exception it gives you back.  And then check the error code.  Using AssertExtension can remove you needing to check for "IsNetNative".

Reply:
`AssertExtension` doesn't contain method overload to examine `WebSocketException` (or `Exception` in general) error message field and return the Exception. Which means, if we use other overload to save the Exception, we still need to use `IsNetNative` check in our test code to verify the ErrorMessage.

1. It's possible to add a method overload inside `AssertExtension` class dedicated for `WebSocketException`, but the overhead is that we need to add `System.Net.WebSockets` reference inside the `CoreFx.private.TestUtilities`, and I'm not sure if that's desirable (Currently that project has no dependency on System.Net).
2. It's not feasible to add a method overload for `Exception` in general that, do both check error message and return the exception. Because all possible distinct function signatures have been taken. Or we will need to add public API (new methods) for that class.

My preference is to use the current if check in our test code, if the desire for checking `WebSocketException` is high, we could consider add it to the common `AssertExtension`.

@davidsh Do you have any thoughts on this?

Reply:
Thx for looking into this.  Let's just keep it the way you're doing it in this PR.